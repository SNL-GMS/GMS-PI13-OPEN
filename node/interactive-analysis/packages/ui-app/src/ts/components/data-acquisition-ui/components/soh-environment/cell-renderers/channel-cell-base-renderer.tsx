import * as React from 'react';
import { dataAcquisitionUIConfig } from '~components/data-acquisition-ui/config';
import {
  getCellStatus,
  getDataReceivedStatus,
  setTooltip
} from '~components/data-acquisition-ui/shared/table/utils';
import { EnvironmentalSoh } from '../types';
const cellHeightPx = dataAcquisitionUIConfig.dataAcquisitionUserPreferences.tableRowHeightPx;
/**
 * Creates a soh-cell with appropriate styles and wraps it around the children
 * of this component.
 * @param environmentalSoh the soh object that is used to create this cell
 */
export const ChannelCellBaseRenderer: React.FunctionComponent<{
  environmentSoh: EnvironmentalSoh;
}> = ({ environmentSoh, children }) => (
  <div
    title={`${setTooltip(environmentSoh)}`}
    style={{
      height: `${cellHeightPx}px`
    }}
    data-cy="soh-cell"
    className={`table-cell soh-cell table-cell__value--numeric`}
    data-cell-status={getCellStatus(environmentSoh?.status, environmentSoh?.isContributing)}
    data-received-status={getDataReceivedStatus(environmentSoh) ?? ''}
  >
    {children}
  </div>
);
