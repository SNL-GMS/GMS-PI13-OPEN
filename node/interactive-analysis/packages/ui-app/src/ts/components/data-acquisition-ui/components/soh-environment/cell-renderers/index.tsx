export { ChannelCellRenderer } from './channel-cell-renderer';
export { MonitorTypeCellRenderer } from './monitor-type-cell-renderer';
