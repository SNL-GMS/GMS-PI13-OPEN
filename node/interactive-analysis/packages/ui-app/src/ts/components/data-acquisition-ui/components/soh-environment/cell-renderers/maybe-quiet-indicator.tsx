import * as React from 'react';
import { QuietIndicator } from '~components/data-acquisition-ui/shared/quiet-indicator';
import { EnvironmentalSoh } from '../types';
import { QuietIndicatorWrapperProps } from './types';

const quietTimingInfoIsDefined = (quietData: EnvironmentalSoh) =>
  quietData &&
  quietData.quietTimingInfo &&
  quietData.quietTimingInfo.quietUntilMs &&
  quietData.quietTimingInfo.quietDurationMs;

/**
 * Renders a quiet indicator if the quietTimingInfo is all there.
 * Renders nothing if no quietUntilMs timestamp has been provided.
 */
export const MaybeQuietIndicator: React.FunctionComponent<QuietIndicatorWrapperProps> = props => {
  if (quietTimingInfoIsDefined(props.data)) {
    return (
      <QuietIndicator
        style={{ diameterPx: props.diameterPx }}
        status={props.data?.status?.toLowerCase()}
        quietTimingInfo={props.data.quietTimingInfo}
        className={props.className}
      />
    );
  }
  return null;
};
