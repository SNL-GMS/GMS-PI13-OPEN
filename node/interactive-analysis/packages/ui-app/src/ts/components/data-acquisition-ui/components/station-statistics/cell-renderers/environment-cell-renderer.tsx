import { NumberCellRendererParams, PercentBar } from '@gms/ui-core-components';
import * as React from 'react';
import { SohRollupCell } from '~components/data-acquisition-ui/shared/table/soh-cell-renderers';
import { CellStatus, formatSohValue } from '~components/data-acquisition-ui/shared/table/utils';
import { getCellStatus, getDataReceivedStatus, setTooltip } from '../../../shared/table/utils';
import { StationStatisticsTableDataContext } from '../types';
import { StationStatisticsDragCell } from './station-statistics-drag-cell';

/**
 * Creates an environment cell, including percent bar child
 */
const EnvironmentCellRenderer: React.FunctionComponent<NumberCellRendererParams> = props => (
  <StationStatisticsTableDataContext.Consumer>
    {context => {
      const data = context.data.find(d => d.id === props.data.id);
      if (!data) return null;

      const stationName = data.stationData?.stationName;
      const dataReceivedStatus = getDataReceivedStatus(data.stationEnvironment);

      // If it is station environment, set to that value; otherwise, use channel environment
      const titleToUse = props.isStationCell
        ? data.stationEnvironment?.toString()
        : setTooltip(data.channelEnvironment);

      // If it is station environment, it is non-contributing; otherwise, figure out channel rollup
      const cellStatusToUse = props.isStationCell
        ? CellStatus.NON_CONTRIBUTING
        : getCellStatus(data.channelEnvironment?.status, data.channelEnvironment?.isContributing);

      // If it is station environment, set to data received status above; otherwise, figure out channel status
      const dataReceivedStatusToUse = props.isStationCell
        ? dataReceivedStatus
        : getDataReceivedStatus(data.channelEnvironment);

      // If it is station environment, format stationEnvironment; otherwise, format channel value
      const valueToUse = props.isStationCell
        ? formatSohValue(data.stationEnvironment)
        : formatSohValue(data.channelEnvironment?.value);

      // If it is station environment, set stationEnvironment percentage; otherwise, use channel percentage
      const percentageToUse = props.isStationCell
        ? data.stationEnvironment
        : data.channelEnvironment?.value;

      return (
        <StationStatisticsDragCell stationId={stationName}>
          <div title={`${titleToUse}`} data-cy="environment-cell">
            <SohRollupCell
              {...props}
              className={`table-cell--numeric`}
              cellStatus={cellStatusToUse}
              dataReceivedStatus={dataReceivedStatusToUse}
              stationId={`${stationName}`}
              value={valueToUse}
            >
              <PercentBar percentage={percentageToUse} />
            </SohRollupCell>
          </div>
        </StationStatisticsDragCell>
      );
    }}
  </StationStatisticsTableDataContext.Consumer>
);

export const ChannelEnvironmentCellRenderer: React.FunctionComponent<NumberCellRendererParams> = props => (
  <EnvironmentCellRenderer {...props} isStationCell={false} />
);

export const StationEnvironmentCellRenderer: React.FunctionComponent<NumberCellRendererParams> = props => (
  <EnvironmentCellRenderer {...props} isStationCell={true} />
);
