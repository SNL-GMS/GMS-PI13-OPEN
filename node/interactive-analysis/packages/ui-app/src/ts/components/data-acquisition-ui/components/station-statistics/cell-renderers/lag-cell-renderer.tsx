import { NumberCellRendererParams } from '@gms/ui-core-components';
import * as React from 'react';
import { SohRollupCell } from '~components/data-acquisition-ui/shared/table/soh-cell-renderers';
import { CellStatus, formatSohValue } from '~components/data-acquisition-ui/shared/table/utils';
import { getCellStatus, getDataReceivedStatus, setTooltip } from '../../../shared/table/utils';
import { StationStatisticsTableDataContext } from '../types';
import { StationStatisticsDragCell } from './station-statistics-drag-cell';

/**
 * Creates a lag cell as a solid cell
 */
const LagCellRenderer: React.FunctionComponent<NumberCellRendererParams> = props => (
  <StationStatisticsTableDataContext.Consumer>
    {context => {
      const data = context.data?.find(d => d.id === props.data?.id);
      if (!data) return null;

      const stationName = data.stationData?.stationName;
      const dataReceivedStatus = getDataReceivedStatus(data.stationLag);

      // If it is station lag, set to that value; otherwise, use channel lag
      const titleToUse = props.isStationCell
        ? data.stationLag?.toString()
        : setTooltip(data.channelLag);

      // If it is station lag, it is non-contributing; otherwise, figure out channel rollup
      const cellStatusToUse = props.isStationCell
        ? CellStatus.NON_CONTRIBUTING
        : getCellStatus(data.channelLag?.status, data.channelLag?.isContributing);

      // If it is station lag, set to data received status above; otherwise, figure out channel status
      const dataReceivedStatusToUse = props.isStationCell
        ? dataReceivedStatus
        : getDataReceivedStatus(data.channelLag);

      // If it is station lag, format stationLag; otherwise, format channel value
      const valueToUse = props.isStationCell
        ? formatSohValue(data.stationLag)
        : formatSohValue(data.channelLag?.value);

      return (
        <StationStatisticsDragCell stationId={stationName}>
          <div title={`${titleToUse}`} data-cy="lag-cell">
            <SohRollupCell
              {...props}
              className={`
                soh-cell--solid
                table-cell--numeric`}
              cellStatus={cellStatusToUse}
              dataReceivedStatus={dataReceivedStatusToUse}
              stationId={`${stationName}`}
              value={valueToUse}
            />
          </div>
        </StationStatisticsDragCell>
      );
    }}
  </StationStatisticsTableDataContext.Consumer>
);

export const ChannelLagCellRenderer: React.FunctionComponent<NumberCellRendererParams> = props => (
  <LagCellRenderer {...props} isStationCell={false} />
);

export const StationLagCellRenderer: React.FunctionComponent<NumberCellRendererParams> = props => (
  <LagCellRenderer {...props} isStationCell={true} />
);
