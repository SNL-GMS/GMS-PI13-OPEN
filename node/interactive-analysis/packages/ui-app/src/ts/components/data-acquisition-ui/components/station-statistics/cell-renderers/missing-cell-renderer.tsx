import { NumberCellRendererParams, PercentBar } from '@gms/ui-core-components';
import * as React from 'react';
import { SohRollupCell } from '~components/data-acquisition-ui/shared/table/soh-cell-renderers';
import { CellStatus, formatSohValue } from '~components/data-acquisition-ui/shared/table/utils';
import { getCellStatus, getDataReceivedStatus, setTooltip } from '../../../shared/table/utils';
import { StationStatisticsTableDataContext } from '../types';
import { StationStatisticsDragCell } from './station-statistics-drag-cell';

/**
 * Creates a missing cell, including percent bar child
 */
const MissingCellRenderer: React.FunctionComponent<NumberCellRendererParams> = props => (
  <StationStatisticsTableDataContext.Consumer>
    {context => {
      const data = context.data.find(d => d.id === props.data.id);
      if (!data) return null;

      const stationName = data.stationData?.stationName;
      const dataReceivedStatus = getDataReceivedStatus(data.stationMissing);

      // If it is station missing, set to that value; otherwise, use channel missing
      const titleToUse = props.isStationCell
        ? data.stationMissing?.toString()
        : setTooltip(data.channelMissing);

      // If it is station missing, it is non-contributing; otherwise, figure out channel rollup
      const cellStatusToUse = props.isStationCell
        ? CellStatus.NON_CONTRIBUTING
        : getCellStatus(data.channelMissing?.status, data.channelMissing?.isContributing);

      // If it is station missing, set to data received status above; otherwise, figure out channel status
      const dataReceivedStatusToUse = props.isStationCell
        ? dataReceivedStatus
        : getDataReceivedStatus(data.channelMissing);

      // If it is station missing, format stationMissing; otherwise, format channel value
      const valueToUse = props.isStationCell
        ? formatSohValue(data.stationMissing)
        : formatSohValue(data.channelMissing?.value);

      // If it is station missing, set stationMissing percentage; otherwise, use channel percentage
      const percentageToUse = props.isStationCell
        ? data.stationMissing
        : data.channelMissing?.value;

      return (
        <StationStatisticsDragCell stationId={stationName}>
          <div title={`${titleToUse}`} data-cy="missing-cell">
            <SohRollupCell
              {...props}
              className={`table-cell--numeric`}
              cellStatus={cellStatusToUse}
              dataReceivedStatus={dataReceivedStatusToUse}
              stationId={`${stationName}`}
              value={valueToUse}
            >
              <PercentBar percentage={percentageToUse} />
            </SohRollupCell>
          </div>
        </StationStatisticsDragCell>
      );
    }}
  </StationStatisticsTableDataContext.Consumer>
);

export const ChannelMissingCellRenderer: React.FunctionComponent<NumberCellRendererParams> = props => (
  <MissingCellRenderer {...props} isStationCell={false} />
);

export const StationMissingCellRenderer: React.FunctionComponent<NumberCellRendererParams> = props => (
  <MissingCellRenderer {...props} isStationCell={true} />
);
