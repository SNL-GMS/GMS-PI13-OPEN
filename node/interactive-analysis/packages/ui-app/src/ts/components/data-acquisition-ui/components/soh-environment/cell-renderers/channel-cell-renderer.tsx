import { PercentBar } from '@gms/ui-core-components';
import * as React from 'react';
import { dataAcquisitionUserPreferences } from '~components/data-acquisition-ui/config';
import { ChannelCellRendererParams, EnvironmentalSoh, EnvironmentTableDataContext } from '../types';
import { ChannelCellBaseRenderer } from './channel-cell-base-renderer';
import { EnvironmentCellValue } from './environment-cell-value';
import { MaybeQuietIndicator } from './maybe-quiet-indicator';

const diameterPx = Math.round(dataAcquisitionUserPreferences.tableRowHeightPx / 3);

/**
 * Renders a `div` around the channel cell which is used to style the cell to indicate that it is selected or not.
 * @param isSelected true if the cell should be marked as selected; false otherwise
 */
const SelectedCellRenderer: React.FunctionComponent<{
  environmentSoh: EnvironmentalSoh;
}> = React.memo(props => {
  // determine if the cell is selected
  const isSelected: boolean = props.environmentSoh?.isSelected;
  return <div className={isSelected ? 'is-selected' : ''}>{props.children}</div>;
});

/**
 * Renders an environment table cell either with data or unknown if no data exists
 */
export const ChannelCellRenderer: React.FunctionComponent<ChannelCellRendererParams> = React.memo(
  props => (
    <EnvironmentTableDataContext.Consumer>
      {context => {
        const data = context.data.find(d => d.id === props.data.id);
        const environmentSoh = data?.valueAndStatusByChannelName.get(props.colDef.colId);
        return (
          <div data-cy="channel-cell">
            <SelectedCellRenderer environmentSoh={environmentSoh}>
              <ChannelCellBaseRenderer environmentSoh={environmentSoh}>
                <MaybeQuietIndicator data={environmentSoh} diameterPx={diameterPx} />
                <PercentBar percentage={environmentSoh?.value} />
                <EnvironmentCellValue
                  value={environmentSoh?.value}
                  hasUnacknowledgedChanges={environmentSoh?.hasUnacknowledgedChanges}
                />
              </ChannelCellBaseRenderer>
            </SelectedCellRenderer>
          </div>
        );
      }}
    </EnvironmentTableDataContext.Consumer>
  )
);
