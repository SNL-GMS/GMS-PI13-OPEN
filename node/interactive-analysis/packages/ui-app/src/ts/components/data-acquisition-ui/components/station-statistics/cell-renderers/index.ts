import {
  ChannelEnvironmentCellRenderer,
  StationEnvironmentCellRenderer
} from './environment-cell-renderer';
import { ChannelLagCellRenderer, StationLagCellRenderer } from './lag-cell-renderer';
import { ChannelMissingCellRenderer, StationMissingCellRenderer } from './missing-cell-renderer';
import { StationNameCellRenderer } from './station-cell-renderer';
import {
  ChannelTimelinessCellRenderer,
  StationTimelinessCellRenderer
} from './timeliness-cell-renderer';
export {
  ChannelTimelinessCellRenderer,
  StationTimelinessCellRenderer,
  ChannelEnvironmentCellRenderer,
  StationEnvironmentCellRenderer,
  ChannelLagCellRenderer,
  StationLagCellRenderer,
  ChannelMissingCellRenderer,
  StationMissingCellRenderer,
  StationNameCellRenderer
};
