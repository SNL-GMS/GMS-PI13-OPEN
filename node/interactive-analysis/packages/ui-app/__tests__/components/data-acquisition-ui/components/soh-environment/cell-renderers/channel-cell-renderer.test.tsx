import { SohTypes } from '@gms/common-graphql';
import React from 'react';
import { ChannelCellRenderer } from '../../../../../../src/ts/components/data-acquisition-ui/components/soh-environment/cell-renderers/channel-cell-renderer';
// tslint:disable-next-line: max-line-length
import {
  EnvironmentalSoh,
  EnvironmentTableDataContext
} from '../../../../../../src/ts/components/data-acquisition-ui/components/soh-environment/types';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

describe("channel cell renderer's", () => {
  const valueAndStatusByChannelNameVal: EnvironmentalSoh = {
    value: 10,
    status: SohTypes.SohStatusSummary.GOOD,
    monitorTypes: SohTypes.SohMonitorType.MISSING,
    channelName: 'channelName',
    hasUnacknowledgedChanges: true,
    quietTimingInfo: {
      quietUntilMs: null,
      quietDurationMs: 100000
    },
    isSelected: false,
    isContributing: true
  };
  const valueAndStatusByChannelName: Map<string, EnvironmentalSoh> = new Map();
  valueAndStatusByChannelName.set('channelName', valueAndStatusByChannelNameVal);

  const myProps: any = {
    colDef: {
      headerName: 'channelName',
      colId: 'channelName'
    },
    context: {
      selectedChannelMonitorPairs: [
        {
          channelName: 'channelName',
          monitorType: 'monitorType'
        }
      ]
    },
    setTooltipProps: jest.fn(),
    rowIndex: 1,
    columnId: 'channelName',
    parentCellHeight: 1,
    parentCellWidth: 1,
    data: {
      id: 'id',
      monitorType: SohTypes.SohMonitorType.LAG,
      monitorStatus: SohTypes.SohStatusSummary.GOOD,
      valueAndStatusByChannelName
    }
  };
  const channelCell = Enzyme.mount(
    <EnvironmentTableDataContext.Provider value={{ data: [myProps.data] }}>
      <ChannelCellRenderer {...myProps} />
    </EnvironmentTableDataContext.Provider>
  );

  // tslint:disable-next-line: prefer-const
  let badProps = JSON.parse(JSON.stringify(myProps));
  badProps.colDef.headerName = 'badChannel';
  badProps.data.valueAndStatusByChannelName = valueAndStatusByChannelName;
  const channelCellNoData = Enzyme.mount(
    <EnvironmentTableDataContext.Provider value={{ data: [badProps.data] }}>
      <ChannelCellRenderer {...badProps} />
    </EnvironmentTableDataContext.Provider>
  );

  it('should be defined', () => {
    expect(ChannelCellRenderer).toBeDefined();
  });
  it('should match snapshot', () => {
    expect(channelCell).toMatchSnapshot();
  });
  it('should match snapshot with bad data', () => {
    expect(channelCellNoData).toMatchSnapshot();
  });
});
