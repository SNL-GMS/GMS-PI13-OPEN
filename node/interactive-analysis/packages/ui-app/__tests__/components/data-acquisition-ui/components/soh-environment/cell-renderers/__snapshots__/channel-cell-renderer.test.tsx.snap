// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channel cell renderer's should match snapshot 1`] = `
<Memo()
  colDef={
    Object {
      "colId": "channelName",
      "headerName": "channelName",
    }
  }
  columnId="channelName"
  context={
    Object {
      "selectedChannelMonitorPairs": Array [
        Object {
          "channelName": "channelName",
          "monitorType": "monitorType",
        },
      ],
    }
  }
  data={
    Object {
      "id": "id",
      "monitorStatus": "GOOD",
      "monitorType": "LAG",
      "valueAndStatusByChannelName": Map {
        "channelName" => Object {
          "channelName": "channelName",
          "hasUnacknowledgedChanges": true,
          "isContributing": true,
          "isSelected": false,
          "monitorTypes": "MISSING",
          "quietTimingInfo": Object {
            "quietDurationMs": 100000,
            "quietUntilMs": null,
          },
          "status": "GOOD",
          "value": 10,
        },
      },
    }
  }
  parentCellHeight={1}
  parentCellWidth={1}
  rowIndex={1}
  setTooltipProps={[MockFunction]}
>
  <div
    data-cy="channel-cell"
  >
    <Memo()
      environmentSoh={
        Object {
          "channelName": "channelName",
          "hasUnacknowledgedChanges": true,
          "isContributing": true,
          "isSelected": false,
          "monitorTypes": "MISSING",
          "quietTimingInfo": Object {
            "quietDurationMs": 100000,
            "quietUntilMs": null,
          },
          "status": "GOOD",
          "value": 10,
        }
      }
    >
      <div
        className=""
      >
        <Component
          environmentSoh={
            Object {
              "channelName": "channelName",
              "hasUnacknowledgedChanges": true,
              "isContributing": true,
              "isSelected": false,
              "monitorTypes": "MISSING",
              "quietTimingInfo": Object {
                "quietDurationMs": 100000,
                "quietUntilMs": null,
              },
              "status": "GOOD",
              "value": 10,
            }
          }
        >
          <div
            className="table-cell soh-cell table-cell__value--numeric"
            data-cell-status="good"
            data-cy="soh-cell"
            data-received-status="received"
            style={
              Object {
                "height": "36px",
              }
            }
            title=""
          >
            <Component
              data={
                Object {
                  "channelName": "channelName",
                  "hasUnacknowledgedChanges": true,
                  "isContributing": true,
                  "isSelected": false,
                  "monitorTypes": "MISSING",
                  "quietTimingInfo": Object {
                    "quietDurationMs": 100000,
                    "quietUntilMs": null,
                  },
                  "status": "GOOD",
                  "value": 10,
                }
              }
              diameterPx={12}
            />
            <PercentBar
              percentage={10}
            >
              <div
                className="percent-bar"
                key="1"
                style={
                  Object {
                    "transform": "translateX(-90%)",
                    "width": "100%",
                  }
                }
              />
            </PercentBar>
            <Component
              hasUnacknowledgedChanges={true}
              value={10}
            >
              <div
                className="table-cell__value table-cell__value--numeric"
                data-cy="environment-cell"
              >
                <Blueprint3.Icon
                  className="env-dirty-dot"
                  icon="symbol-circle"
                >
                  <span
                    className="bp3-icon bp3-icon-symbol-circle env-dirty-dot"
                    icon="symbol-circle"
                  >
                    <svg
                      data-icon="symbol-circle"
                      height={16}
                      viewBox="0 0 16 16"
                      width={16}
                    >
                      <desc>
                        symbol-circle
                      </desc>
                      <path
                        d="M8 3.01a5 5 0 100 10 5 5 0 100-10z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </span>
                </Blueprint3.Icon>
                <span>
                  10.0
                </span>
              </div>
            </Component>
          </div>
        </Component>
      </div>
    </Memo()>
  </div>
</Memo()>
`;

exports[`channel cell renderer's should match snapshot with bad data 1`] = `
<Memo()
  colDef={
    Object {
      "colId": "channelName",
      "headerName": "badChannel",
    }
  }
  columnId="channelName"
  context={
    Object {
      "selectedChannelMonitorPairs": Array [
        Object {
          "channelName": "channelName",
          "monitorType": "monitorType",
        },
      ],
    }
  }
  data={
    Object {
      "id": "id",
      "monitorStatus": "GOOD",
      "monitorType": "LAG",
      "valueAndStatusByChannelName": Map {
        "channelName" => Object {
          "channelName": "channelName",
          "hasUnacknowledgedChanges": true,
          "isContributing": true,
          "isSelected": false,
          "monitorTypes": "MISSING",
          "quietTimingInfo": Object {
            "quietDurationMs": 100000,
            "quietUntilMs": null,
          },
          "status": "GOOD",
          "value": 10,
        },
      },
    }
  }
  parentCellHeight={1}
  parentCellWidth={1}
  rowIndex={1}
>
  <div
    data-cy="channel-cell"
  >
    <Memo()
      environmentSoh={
        Object {
          "channelName": "channelName",
          "hasUnacknowledgedChanges": true,
          "isContributing": true,
          "isSelected": false,
          "monitorTypes": "MISSING",
          "quietTimingInfo": Object {
            "quietDurationMs": 100000,
            "quietUntilMs": null,
          },
          "status": "GOOD",
          "value": 10,
        }
      }
    >
      <div
        className=""
      >
        <Component
          environmentSoh={
            Object {
              "channelName": "channelName",
              "hasUnacknowledgedChanges": true,
              "isContributing": true,
              "isSelected": false,
              "monitorTypes": "MISSING",
              "quietTimingInfo": Object {
                "quietDurationMs": 100000,
                "quietUntilMs": null,
              },
              "status": "GOOD",
              "value": 10,
            }
          }
        >
          <div
            className="table-cell soh-cell table-cell__value--numeric"
            data-cell-status="good"
            data-cy="soh-cell"
            data-received-status="received"
            style={
              Object {
                "height": "36px",
              }
            }
            title=""
          >
            <Component
              data={
                Object {
                  "channelName": "channelName",
                  "hasUnacknowledgedChanges": true,
                  "isContributing": true,
                  "isSelected": false,
                  "monitorTypes": "MISSING",
                  "quietTimingInfo": Object {
                    "quietDurationMs": 100000,
                    "quietUntilMs": null,
                  },
                  "status": "GOOD",
                  "value": 10,
                }
              }
              diameterPx={12}
            />
            <PercentBar
              percentage={10}
            >
              <div
                className="percent-bar"
                key="2"
                style={
                  Object {
                    "transform": "translateX(-90%)",
                    "width": "100%",
                  }
                }
              />
            </PercentBar>
            <Component
              hasUnacknowledgedChanges={true}
              value={10}
            >
              <div
                className="table-cell__value table-cell__value--numeric"
                data-cy="environment-cell"
              >
                <Blueprint3.Icon
                  className="env-dirty-dot"
                  icon="symbol-circle"
                >
                  <span
                    className="bp3-icon bp3-icon-symbol-circle env-dirty-dot"
                    icon="symbol-circle"
                  >
                    <svg
                      data-icon="symbol-circle"
                      height={16}
                      viewBox="0 0 16 16"
                      width={16}
                    >
                      <desc>
                        symbol-circle
                      </desc>
                      <path
                        d="M8 3.01a5 5 0 100 10 5 5 0 100-10z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </span>
                </Blueprint3.Icon>
                <span>
                  10.0
                </span>
              </div>
            </Component>
          </div>
        </Component>
      </div>
    </Memo()>
  </div>
</Memo()>
`;
